libexec/python/docker/api.py

```diff
@@ -163,19 +163,27 @@ def update_token(self, response=None, auth=None):
                 sys.exit(1)
 
             challenge = response.headers["Www-Authenticate"]
-            regexp = '^Bearer\s+realm="(.+)",service="(.+)",scope="(.+)",?'
+            regexp = '^Bearer\s+realm="(.+?)"'
```

Comment:
Do you have any spec where it states the order shall always be realm, service, scope? In the absence of something like that I can't help thinking the order shouldn't matter in matching those. See comment further down also...

Reply:
I don't have a spec and also don't believe the order matters - I simply kept the ordering because that's how it was before and wanted to make the patch lean for review. I mentioned in the description of #1184 that there are libs out there that provide more robust `Www-Authenticate` challenge parsing, e.g.:

https://github.com/alexsdutton/www-authenticate

I can update the patch to ignore the order of the params in the challenge if you like?

Reply:
That would be great to ignore order if you are happy to do that. Thanks. It'd make me happier that we wouldn't see an issue with a future registry that is sending out things in a slightly different order.

Reply:
Sounds good I'll try to get that in soon.